### PR TITLE
K3s setup

### PR DESCRIPTION
**Pull request by rajibhossen**

### CODE REVIEW

**Comment by vsoch** [comment]
Nice job @rajibhossen! :partying_face:  Two quick questions and a suggested change:

- Is there a reason you want k3s to come up with the entire cluster? I think the use case we were thinking about (at least for the lab) was to have it launched with a job, meaning running in the user's namespace, and it's not running _alongside_ flux but _within_ it (and then multiple users could each control their own cluster). But if you are just doing some testing of how it scales, etc. this should probably be OK.
- Related to the above - do you want to test rootless mode (or not relevant yet?)
- And one suggestion / nit is that "flux-selector" should be a variable that matches the name, see [here](https://github.com/converged-computing/flux-terraform-ami/blob/186f9254c281ede95b01b2cf4b6b3335148d67f4/examples/scripts/flux-setup.sh#L7) and [here](https://github.com/converged-computing/flux-terraform-ami/blob/186f9254c281ede95b01b2cf4b6b3335148d67f4/examples/autoscale/main.tf#L42) for an example of doing that.

**Comment by rajibhossen** [comment]
Thank you!
1. Yes, I'll integrate/work with that later. I wanted to see k3s deployment using the TF and scripts. 
2. I'm not sure about that yet!
3. That's a mistake. I was using k3s_setup.sh script from inside of an instance to test. I'll remove that.

**Comment by vsoch** [comment]
Gotcha! Okay one more request - sorry just noticed the structure. I think Abhik had mentioned this, but please make a separate k3s folder within experiments and alongside the autoscale folder that looks like this:

```
experiments/
   autoscale/
   k3s
```
And don't do any modifications to the autoscale setup. The reason is because this repository is not a single hard coded research project, but a set of separate, modular experiments (each scoped to a particular purpose) and so the one you are working on (and then will have full ownership over!) is this k3s experiment! And someone can come later that _just_ wants Flux with autoscale, and they still are able to do that without needing to untangle the k3s. Does that make sense?

**Comment by rajibhossen** [comment]
I've decoupled the examples.

File: examples/k3s/main.tf [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+# VARIABLES # for you to edit!
+
+locals {
+  name          = "k3s"
+  pwd           = basename(path.cwd)
+  region        = "us-east-1"
+  ami           = "ami-0ff535566e7c13e8c"
+  instance_type = "m4.large"
+  vpc_cidr      = "10.0.0.0/16"
+  key_name      = "rajib_ssh_2"
```
Comment (vsoch) [inline examples/k3s/main.tf]: Do you want to generalize this too (as you did for autoscale) to not reveal your key name? Or maybe a better idea - have it come from the environment?

**Review by vsoch — COMMENTED** [review]
Perfecto! One quick question, and then we are ready to merrrrge away Merrill!  :baseball:

**Review by vsoch — APPROVED** [review]

**Pull request merged**